grails-app/services/org/pih/warehouse/core/DashboardService.groovy

```diff
@@ -175,32 +184,40 @@ class DashboardService {
      * @param threshold the threshold filter
      * @return a list of inventory items
      */
-    List getExpiringStock(Category category, Location location, String expirationStatus) {
+    List getExpiringStock(InventoryReportCommand command) {
         long startTime = System.currentTimeMillis()
 
         def today = new Date()
         today.clearTime()
         // Get all stock expiring ever (we'll filter later)
         def expiringStock = InventoryItem.findAllByExpirationDateGreaterThanEquals(today + 1, [sort: 'expirationDate', order: 'asc'])
-        def quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(location)
+        def quantityMap = productAvailabilityService.getQuantityOnHandByInventoryItem(command.location)
         expiringStock = expiringStock.findAll { quantityMap[it] > 0 }
-        if (category) {
-            expiringStock = expiringStock.findAll { item -> item?.product?.category == category }
+        if (command.category) {
+            expiringStock = expiringStock.findAll { item -> item?.product?.category == command.category }
+        }
+
+        if (command.startDate) {
+            expiringStock = expiringStock.findAll { item -> item?.expirationDate >= command.startDate }
+        }
+
+        if (command.endDate) {
+            expiringStock = expiringStock.findAll { item -> item?.expirationDate <= command.endDate }
```

Comment:
I'm approving this one, but this needs to be refactored asap, especially if the performance is bad there.